src/bg/user-script-registry.js

```diff
@@ -47,59 +47,65 @@ async function openDb() {
 ///////////////////////////////////////////////////////////////////////////////
 
 async function installFromDownloader(userScriptDetails, downloaderDetails) {
+  let remoteScript = new RemoteUserScript(userScriptDetails);
+
   let db = await openDb();
-  return new Promise(async (resolve, reject) => {
-    try {
```

Comment:
What happens to these exceptions that are no longer being caught by anything?

Reply:
I don't quite understand:

`installFromDownloader` itself is async (returns rejected Promise, if error is thrown).
Inside the executor, `reject` is called, then `catch` and `throw` => see above. IIRC.

Reply:
For this particular point there's a total of five statements that would ever get caught in the `try .. catch` block. Three of those statements take static arguments and will not throw [1]. As for the other two, if they do throw the error will be bubbled up the async / promise chain.

If the database operation fails then the returned `new Promise` rejects and the error is again bubbled up the async / promise chain. The same applies with any error that may occur in the `.then` blocks. When using async / promises, there's no need to catch errors if all you're going to do is reject the promise in the catch block [2]. Promise structures do that for you.

[1] This assumes that the static arguments are coded properly and this bars any uncontrollable encounters caused by bad Firefox code (unlikely we could do anything about that anyway).
[2] This only applies if you are not going to change the error that is rejected. If you wanted to reject with a different error than you'd have to catch.

Reply:
A bit of sample code:
```
(async () => {
  throw 'err1';
  return new Promise((resolve, reject) => {
    throw 'err2';
    resolve('OK');
  }).then(() => {
    throw 'err3';
  }).catch((err) => {
    throw 'err4: ' + err;
  });
})();
```
Results (3 cases: 1. whole code, 2. without `err1`, 3. without `err1` and `err2`):
```
/*
Promise (rejected: err1)
*/
/*
Promise (rejected: err4: err2)
*/
/*
Promise (rejected: err4: err3)
*/
```

Reply:
@Eselce could you do ` ```javascript ` for the opening code tag, to get syntax highlighting?

Reply:
@Sxderp :whistle: always forget that... :grin: